docs/idiomatic-maya-guide.md

```diff
@@ -1,99 +1,179 @@
+2. Appropriate to the style of art or music associated with a particular period, individual, or group.
 
 I do not remember hearing anything about being idiomatic for other languages. I guess it has become synonymous with the advent of golang. Being idiomatic definitely means more than just good coding practices. It tries to provide a common ground for each member of the team to communicate in a way that is understood clearly. Obviously this is not solved by programming language, not even a high level programming language. IMO current day programming languages provide a bunch of dialects and lets its users (i.e. developers) choose the one they like the most. The fact that there is a choice, it hits the team hard later in the project's release lifecycle. The project code may not look as bad as sphagetti but no good either to understand fast and hence implement features, fixes, etc. faster.
 
 ## If Maya is Go then why not Idiomatic Go
+
 Being idiomatic in Maya includes all the idiomatic pieces in Go and also takes into consideration writing code that understands and responds to Kubernetes a lot better. In addition, team at Maya has tried to put all their storage learnings so far into what is being termed as Idiomatic Maya where Maya is control plane for OpenEBS.
 
+This document lists some of the guidelines and their corresponding examples that we are trying to follow in Maya to be idiomatic in true sense (there could be difference in opinions but we are trying to make sure that we all remain on the same page and hence it doesn't look confusing going further) -
 
-## Devil is in the details
-Below are some sample code reviews that describes with examples to write idiomatic maya code.
+## Naming Conventions
 
-### Naming - Thumb Rule
-- Use names based on what the logic provides and not based on what the logic contains
+- Try to use names based on what the logic provides and not based on what the logic contains
 
 ```go
 import (
+  // Bad
   // alias `algorithm` conveys about what it contains i.e. some algorithm
   algorithm "github.com/openebs/maya/pkg/algorithm/nodeSelect/v1alpha1"
 
   // vs.
 
+  // Good
   // `nodeselect` tries to convey what the logic provides
   // this seems more natural way to express
   nodeselect "github.com/openebs/maya/pkg/algorithm/nodeSelect/v1alpha1"
 )
 ```
 
 ```go
-nodeDisks, err := ac.NodeDiskSelector()
-// `disks` is repeated; does not seem natural
-if len(nodeDisks.Disks.Items) == 0 {
-  // ...
-}
+import(
+// Bad
+// upgrade looks more generic i.e upgrade package can have
+// different other sub-packages like result which can be imported
+upgrade "github.com/openebs/maya/pkg/upgrade/result/v1alpha1"
+
+// vs
+
+// Good
+// upgraderesult looks more specific and to the point i.e. the alias
+// is for 'result' which is inside upgrade
+upgraderesult "github.com/openebs/maya/pkg/upgrade/result/v1alpha1"
+```
+
+```go
+// Bad
+// taskPatch represents details for runtask patch operation
+type taskPatch struct {}
```

Comment:
This naming depends on package name as well.
So `patch` can be valid as well.

Reply:
Done